source/generators/MuonAngleGenerator.cc

```diff
@@ -110,6 +164,18 @@ void MuonAngleGenerator::SetupAngles()
 void MuonAngleGenerator::GeneratePrimaryVertex(G4Event* event)
 {
 
+  // Initalise RNG seeds and load file on the first event
+  if (event->GetEventID() == 0){ 
```

Comment:
I think we must avoid this clause. Notice that the first event id is not 0 in general, since this is set via `/nexus/persistency/start_id`. What would the seed be in such a case? Could we just take the seed from `/nexus/random_seed`?

Reply:
Right, I couldn't figure out a way to initialize it only once, this protection was added to stop initializing it on every new event. We could possibly use the same protection as the rotation is initialised ie. if ( rPhi_ == NULL) instead if you prefer.  

The CLHEP::HepRandom::getTheSeed() should be already grabbing the /nexus/random_seed configured.

Reply:
or we can add a data member bool to flag if initialised, that is a little clearer actually.

Reply:
Do we really need those `RN_engine`? Can't you just use the functions (e.g. `G4UniformRandom()`) provided by Geant4?

https://geant4.kek.jp/lxr/source/global/HEPRandom/include/Randomize.hh

Reply:
> I think we must avoid this clause. Notice that the first event id is not 0 in general, since this is set via `/nexus/persistency/start_id`. 

Just for the sake of correctness: The first event ID is always 0. We only modify it at the time we write the events in the output file.

Reply:
The random engine is used as an input to the std::discrete_distribution which is what I use to sample the events from. I don't think I can use the G4UniformRandom() for that. 

I can probably replace the other RN generators used for the Gaussian smearing, but I just chose to use the std functions for consistency. Let me know if you would like to use the Geant4 function in this case and I can swap it.